features/messages/impl/src/test/kotlin/io/element/android/features/messages/timeline/factories/event/TimelineItemContentPollFactoryTest.kt

```diff
@@ -0,0 +1,288 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.messages.timeline.factories.event
+
+import com.google.common.truth.Truth
+import io.element.android.features.messages.impl.timeline.factories.event.TimelineItemContentPollFactory
+import io.element.android.features.messages.impl.timeline.model.event.TimelineItemPollContent
+import io.element.android.features.poll.api.PollAnswerItem
+import io.element.android.libraries.featureflag.api.FeatureFlags
+import io.element.android.libraries.featureflag.test.FakeFeatureFlagService
+import io.element.android.libraries.matrix.api.poll.PollAnswer
+import io.element.android.libraries.matrix.api.poll.PollKind
+import io.element.android.libraries.matrix.api.timeline.item.event.PollContent
+import io.element.android.libraries.matrix.test.A_USER_ID
+import io.element.android.libraries.matrix.test.A_USER_ID_10
+import io.element.android.libraries.matrix.test.A_USER_ID_2
+import io.element.android.libraries.matrix.test.A_USER_ID_3
+import io.element.android.libraries.matrix.test.A_USER_ID_4
+import io.element.android.libraries.matrix.test.A_USER_ID_5
+import io.element.android.libraries.matrix.test.A_USER_ID_6
+import io.element.android.libraries.matrix.test.A_USER_ID_7
+import io.element.android.libraries.matrix.test.A_USER_ID_8
+import io.element.android.libraries.matrix.test.A_USER_ID_9
+import io.element.android.libraries.matrix.test.FakeMatrixClient
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+private const val A_POLL_QUESTION = "What is your favorite food?"
+private val A_POLL_ANSWER_1 = PollAnswer("id_1", "Pizza")
+private val A_POLL_ANSWER_2 = PollAnswer("id_2", "Pasta")
+private val A_POLL_ANSWER_3 = PollAnswer("id_3", "French Fries")
+private val A_POLL_ANSWER_4 = PollAnswer("id_4", "Hamburger")
+
+internal class TimelineItemContentPollFactoryTest {
+
+    private val factory = TimelineItemContentPollFactory(
+        matrixClient = FakeMatrixClient(),
+        featureFlagService = FakeFeatureFlagService(mapOf(FeatureFlags.Polls.key to true)),
+    )
+
+    @Test
+    fun `Disclosed poll - not ended states`() = runTest {
+        // No votes
+        Truth.assertThat(
+            factory.create(aPollContent(PollKind.Disclosed))
+        ).isEqualTo(aTimelineItemPollContent(PollKind.Disclosed))
+
+        // Some votes, according one from current user
+        val votes = mapOf(
+            A_POLL_ANSWER_1.id to listOf(A_USER_ID_2, A_USER_ID_3, A_USER_ID_4),
+            A_POLL_ANSWER_2.id to listOf(A_USER_ID, A_USER_ID_5, A_USER_ID_6, A_USER_ID_7, A_USER_ID_8, A_USER_ID_9),
+            A_POLL_ANSWER_3.id to emptyList(),
+            A_POLL_ANSWER_4.id to listOf(A_USER_ID_10),
+        )
+        Truth.assertThat(
+            factory.create(aPollContent(PollKind.Disclosed).copy(votes = votes))
```

Comment:
Done, I tried to factorize a bit. I chose to duplicate the parameters to have a better view of the initial state and the resulting state at the same time.
Let me know if you prefer this format. I could not find a way to deduplicate the expected results since there are some differences on each test case :/